ui/components/multichain/network-list-menu/network-list-menu.js

```diff
@@ -96,18 +96,25 @@ export const NetworkListMenu = ({ onClose }) => {
       return nonTestNetworks;
     }
 
-    // Reorder nonTestNetworks based on the order of chainIds in orderedNetworksList
-    const sortedNetworkList = orderedNetworksList
-      .map((chainId) =>
-        nonTestNetworks.find((network) => network.chainId === chainId),
-      )
-      .filter(Boolean);
-
-    return sortedNetworkList;
+    // Create a mapping of chainId to index in orderedNetworksList
+    const orderedIndexMap = {};
+    orderedNetworksList.forEach((network, index) => {
+      orderedIndexMap[`${network.networkId}_${network.networkRpcUrl}`] = index;
+    });
+
+    // Sort nonTestNetworks based on the order in orderedNetworksList
+    const sortedNonTestNetworks = nonTestNetworks.sort((a, b) => {
+      const keyA = `${a.chainId}_${a.rpcUrl}`;
+      const keyB = `${b.chainId}_${b.rpcUrl}`;
+      return orderedIndexMap[keyA] - orderedIndexMap[keyB];
+    });
+
+    return sortedNonTestNetworks;
```

Comment:
Can this data manipulation be moved to a selector?  It seems like a lot to be doing in side a UI component.

Reply:
@darkwing  I think we can do it in a follow up PR with the e2e test that we talked about today?